src/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -50,13 +105,121 @@ rawPageNumBlobs :: RawPage -> Word16
 rawPageNumBlobs (RawPage off ba) = indexByteArray ba (off + 1)
 
 type KeyOffset = Word16
+type ValueOffset = Word16
 
-rawPageKeyOffsets :: RawPage -> Vector KeyOffset
+rawPageKeyOffsets :: RawPage -> P.Vector KeyOffset
 rawPageKeyOffsets page@(RawPage off ba) =
-    Vector (off + fromIntegral (div2 dirOffset)) (fromIntegral dirNumKeys) ba
+    P.Vector (off + fromIntegral (div2 dirOffset)) (fromIntegral dirNumKeys + 1) ba
```

Comment:
Yes, yes indeed due the cunning overlapping representation, the +1 last vector entryhere is the offset one past the final key.